### DISCUSSION TITLE
Upgrading to 1.7.2 makes intellij hang when sourcing .vimrc

### DISCUSSION THREAD

**Discussion by serpro69** [JetBrains/ideavim#372] [Q&A] [answered]
I've recently upgraded ideavim to 1.7.2 and it seems that this broke things for me even further and made intellij completely unusable as compared to 1.7.0 (where I saw some issues with vim configs, but intellij as a whole was still usable)

My `.ideavimrc` is very simple:
```
set surround
set multiple-cursors
set clipboard+=ideaput

nmap gi :action GotoImplementation<CR>
nmap gs :action GotoSuperMethod<CR>
set ideajoin

" source ~/.vimrc

" IDEA
nnoremap <leader>ga :action GotoAction<cr>
nnoremap <leader>gs :action GotoSymbol<cr>
nnoremap <leader>gf :action GotoFile<cr>
nnoremap <leader>gc :action GotoClass<cr>
nnoremap <leader>gd :action GotoDeclaration<cr>
nnoremap <leader>gi :action GotoImplementation<cr>

nnoremap <leader>ff :action FindInPath<cr>
nnoremap <leader>fu :action FindUsages<cr>
nnoremap <leader>re :action RenameElement<cr>
```

And what breaks things is sourcing the `.vimrc` file (here's the contents - https://gist.github.com/serpro69/3504dcd0c7e7e2e8a7e64c88af04519a )

The issues is, when I open any project now - intellij just hangs on "loading project". If I comment out `source ~/.vimrc` in the .ideavimrc, then intellij works just fine.
Anyone can help me figure out what's wrong and how I could fix it?

**Comment by ngnmhieu** [comment]
Same for me. My temporary solution is to copy necessary configurations and mappings into .ideavimrc file.

**Reply by yongxingzhao** [reply]
I also encountered the same problem, and the solution was the same as yours.

**Reply by AlexPl292** [reply]
Folks, could you please attach your `~/.ideavimrc` and `.vimrc` configurations?
@lippfi

**Reply by yongxingzhao** [reply]
> Folks, could you please attach your `~/.ideavimrc` and `.vimrc` configurations? @lippfi

ideavimrc：https://github.com/YongXingZhao/configuration/blob/main/.ideavimrc
vimrc：https://github.com/YongXingZhao/configuration/blob/main/.vimrc

**Reply by lippfi** [reply]
Sorry, parser is super slow for large configurations. We plan to significantly increase parsing speed in IdeaVim 1.8 (EAP is planned for the next week).
As a temporary solution you can follow @ngnmhieu 's advice or wrap unnecessary for IdeaVim lines with following comment:
```
"ideaVim ignore
call unusedCommand()
...
"ideaVim ignore end
```

**Reply by yongxingzhao** [reply]
> Sorry, parser is super slow for large configurations. We plan to significantly increase parsing speed in IdeaVim 1.8 (EAP is planned for the next week). As a temporary solution you can follow @ngnmhieu 's advice or wrap unnecessary for IdeaVim lines with following comment:
> 
> ```
> "ideaVim ignore
> call unusedCommand()
> ...
> "ideaVim ignore end
> ```

Ok,Thank you

**Comment by serpro69** [comment, marked as answer]
Performance has been improved with 1.8.0 and ideavim is usable again.
Thanks :) I'll mark this as answered.
